lib/manager/maven/readme.md

```diff
@@ -0,0 +1,138 @@
+## Overview
+
+#### Name of package manager
+
+Maven
+
+---
+
+#### What language does this support?
+
+Java
+
+---
+
+#### Does that language have other (competing?) package managers?
+
+Gradle 
+
+## Package File Detection
+
+#### What type of package files and names does it use?
+
+"pomfile" usually named pom.xml. It is possible to override default pom.xml with any filename
+
+ `mvn -f <path>/custom_pom.xml <command>`
+
+---
+
+#### What [fileMatch](https://renovatebot.com/docs/configuration-options/#filematch) pattern(s) should be used?
+
+`['(^|/)([\\w-]*)pom.xml$']`
+
+---
+
+#### Is it likely that many users would need to extend this pattern for custom file names?
+
+Not many but some user might use custom filenames.
+
+---
+
+#### Is the fileMatch pattern likely to get many "false hits" for files that have nothing to do with package management?
+
+Usually "pomfile" is in the root of the project and it is not likely that it will be many other xml files in the same directory.
+
+## Parsing and Extraction
+
+#### Can package files have "local" links to each other that need to be resolved?
+
+No
+
+#### Is there reason why package files need to be parsed together (in serial) instead of independently?
+
+No
+
+---
+
+#### What format/syntax is the package file in? e.g. JSON, TOML, custom?
+
+XML
+
+---
+
+#### How do you suggest parsing the file? Using an off-the-shelf parser, using regex, or can it be custom-parsed line by line?
+
+Any XML reader.
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+No
+
+---
+
+#### List all the sources/syntaxes of dependencies that can be extracted:
```

Comment:
If there's only one way to define dependency versions, can you give a code (XML) example here?

Reply:
`<dependency>
            <groupId>org.springframework.integration</groupId>
            <artifactId>spring-integration-aws</artifactId>
            <version>1.2.3</version>
        </dependency>`

Reply:
So would a range specification also be valid there, e.g. `<version>(,1.0],[1.2,)</version>`?

Reply:
yes